shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -167,12 +169,21 @@ - (void)setViewController:(FlutterViewController*)viewController {
   _viewController = [viewController getWeakPtr];
   self.iosPlatformView->SetOwnerViewController(_viewController);
   [self maybeSetupPlatformViewChannels];
+
+  self.flutterViewControllerWillDeallocObserver =
```

Comment:
I'm not doing that because that would be changing behavior outside the bounds of the problem I was fixing.  I'm just moving this file over to the new mechanism I created so we don't have multiple ways to do the same thing.  The only logical change I made here was calling `_viewController.reset()` in `notifyViewControllerDeallocated` . I didn't technically have to do that to solve the bug, but it will almost certainly cause a bug in a future to keep a weak pointer to a dealloced object.  This will turn that future bug to a null pointer exception which will be easier for someone to fix than calling a method into random memory.

Reply:
Discussed offline. Moving the direct plumbing to NSNotificationCenter is ok. If we're implicitly saying the precedent is also to promote moving other event plumbing to notification center.